### PR TITLE
CASMHMS-6148: Parse PowerConsumedWatts for any data type and intialize pcap min/max appropriately

### PR DESCRIPTION
**Pull request by jwlv**
## Summary and Scope

The Paradise redfish Power endpoint presents the consumed watts value as a float. All prior platforms have presented it as an integer value and thus the PCS data structure for holding it declared it as an int. This caused json.Unmarshal() to fail on Paradise because it enforces strict type checking.

The fix here is to change the int type to an interface{} type so that it can hold a value of any type. We then convert any floats we read to an int (by rounding). We chose to do this rather than simply changing the type from an int to a float so that we don't break any customer code that may be doing the same thing we were doing (expecting an int rather than a float).

Additionally, this PR contains a fix to initialize the power cap min/max values to -1.  This is necessary in order for requests against non-oem platforms to work properly in doPowerCapTask() where checks are made against these values.

## Issues and Related PRs

* Resolves CASMHMS-6148

## Testing

Tested on:

  * tyr

Test description:

With a patched SMD in place, verified that PCS would convert floats to ints properly when ran against Paradise nodes.  Then verified that PCS could power cap both Paradise and Bard Peak compute nodes.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? Y
- Were continuous integration tests run? If not, why? Y
- Was upgrade tested? If not, why? Y
- Was downgrade tested? If not, why? Y

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: internal/domain/power-cap.go [hunk 1]
```diff
@@ -732,6 +734,23 @@ func doPowerCapTask(taskID uuid.UUID) {
...
+
+				// Convert PowerConsumedWatts to an int if not already (it's an interface{}
+				// type that can support ints and floats) - Needed for Foxconn Paradise,
+				// perhaps others in the future
+				for _, pwrCtl := range rfPower.PowerCtl {
+					if pwrCtl.PowerConsumedWatts != nil {
+						switch v := (*pwrCtl.PowerConsumedWatts).(type) {
+						case float64:	// Convert to int
+							*pwrCtl.PowerConsumedWatts = int(math.Round(v))
+							logger.Log.WithFields(logrus.Fields{"type": reflect.TypeOf(*pwrCtl.PowerConsumedWatts), "value": *pwrCtl.PowerConsumedWatts}).Errorf("Unexpected type/value detected for PowerConsumedWatts, setting to 0\n")
```
Comment (shunr-hpe) [inline internal/domain/power-cap.go:746]: Bad log statement. Copy and paste from the default case log.

File: internal/hsm/state_manager.go [hunk 2]
```diff
@@ -1,17 +1,17 @@
 // MIT License
-// 
-// (C) Copyright [2022-2023] Hewlett Packard Enterprise Development LP
-// 
+//
+// (C) Copyright [2022-2023,2025] Hewlett Packard Enterprise Development LP
```
Comment (shunr-hpe) [inline internal/hsm/state_manager.go:3]: 2024

**Review by shunr-hpe — APPROVED** [review]

File: internal/domain/power-cap.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright [2021-2024] Hewlett Packard Enterprise Development LP
+ * (C) Copyright [2021-2025] Hewlett Packard Enterprise Development LP
```
Comment (mbuchmann-hpe) [inline internal/domain/power-cap.go:2]: Pushing it a little too much.

File: internal/hsm/state_manager.go (hunk 2 above)
Comment (mbuchmann-hpe) [inline internal/hsm/state_manager.go:3]: Now your making me feel old... Y2k was 25 years ago.

**Review by mbuchmann-hpe — APPROVED** [review]

File: internal/domain/power-cap.go (hunk 1 above)
Comment (jwlv) [reply internal/domain/power-cap.go]: Good catch!  I had copy/pasted it there to make sure the default case log would log correctly if/when it ever gets hit.  I've pushed a change to resolve this.

File: internal/hsm/state_manager.go (hunk 2 above)
Comment (jwlv) [reply internal/hsm/state_manager.go]: Wow, what's wrong with my clock?!  I pushed a change to resolve this.

File: internal/domain/power-cap.go (hunk 3 above)
Comment (jwlv) [reply internal/domain/power-cap.go]: Resolved.

File: internal/hsm/state_manager.go (hunk 2 above)
Comment (jwlv) [reply internal/hsm/state_manager.go]: I pushed a change to resolve this.

**Pull request merged**